apps/demo/src/pages/experimental/resolver.astro

```diff
@@ -0,0 +1,49 @@
+---
+import type { Submit } from "@astro-reactive/common";
+import Form, { FormGroup } from "@astro-reactive/form";
+import { z } from "zod";
+import Layout from "../../components/Layout.astro";
+import { zodResolver } from "@astro-reactive/resolvers";
+
+const zodSchema = z.object({
+  name: z.string(),
+  email: z.string().email(),
+  age: z.number().optional(),
+  familySize: z.any(),
+});
+
+type FieldValues = z.infer<typeof zodSchema>;
+
+const formGroup = new FormGroup<FieldValues>(
```

Comment:
I'm thinking of a way so that FormGroup will be able to fully infer types inside its constructor. If that is possible, we remove the need for line 15 and this optional `<FieldValues>`

Rough idea is that the FormGroup constructor will check if there is a `resolver` property in the formConfig (2nd argument of the constructor), and then can we use the keys of the object there to strictly type the 1st argument.

Let me know what you think!

Reply:
It's possible to do that, but I think the solution with optional generic enable users who don't use `resolver` to have types. 
 
I will still try and update you on the implementation later.